crates/ty_python_semantic/src/types.rs

```diff
@@ -6140,11 +6288,62 @@ impl<'db> TypeVarInstance<'db> {
             self.definition(db),
             self.bound_or_constraints(db)
                 .map(|b| b.materialize(db, variance)),
-            self.variance(db),
+            self.explicit_variance(db), // TODO: this could be Some(self.inferred_variance(db))
             self.default_ty(db),
             self.kind(db),
         )
     }
+
+    #[track_caller]
```

Comment:
I don't think we want `track_caller` here. That's intended more for e.g. assertion helpers, where locations within the function aren't useful to include in a panic message or stack trace